ui/src/App.js

```diff
@@ -0,0 +1,135 @@
+import { Calendar, momentLocalizer } from "react-big-calendar";
+import moment from "moment";
+import "react-big-calendar/lib/css/react-big-calendar.css";
+import { useState } from "react";
+import s from "./App.module.css";
+
+const localizer = momentLocalizer(moment);
+const hours = new Date().getHours();
+const minutes = new Date().getMinutes();
+const padNumberWith0 = (num) => num.toString().padStart(2, "0");
+const defaultStartTime = `${padNumberWith0(hours)}:${padNumberWith0(minutes)}`;
+
+Date.prototype.addHours = function (h) {
+  this.setTime(this.getTime() + h * 60 * 60 * 1000);
+  return this;
+};
+
+const defaultEndTime = `${padNumberWith0(hours + 1)}:${padNumberWith0(
+  minutes
+)}`;
+
+function padTo2Digits(num) {
+  return num.toString().padStart(2, "0");
+}
+
+function formatDate(date) {
+  return [
+    date.getFullYear(),
+    padTo2Digits(date.getMonth() + 1),
+    padTo2Digits(date.getDate()),
+  ].join("-");
+}
+
+const formatAMPM = (timeString) => {
+  const date = timeString.substring(0, 5);
+  let hours = date.substring(0, 2);
+  let minutes = date.substring(3, 5);
+  let ampm = hours >= 12 ? "pm" : "am";
+  hours = hours % 12;
+  hours = hours ? hours : 12;
+  let strTime = hours + ":" + minutes + " " + ampm;
+  return strTime;
+};
+
+function App() {
+  const [startDate, setStartDate] = useState(formatDate(new Date()));
+  const [endDate, setEndDate] = useState(formatDate(new Date()));
+  const [startTime, setStartTime] = useState(defaultStartTime);
+  const [endTime, setEndTime] = useState(defaultEndTime);
+  const [title, setTitle] = useState("");
+  const [events, setEvents] = useState([]);
+  const [error, setError] = useState(null);
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    const startDateAndTime = new Date(`${startDate}T${startTime}`);
+    const endDateAndTime = new Date(`${endDate}T${endTime}`);
+
+    if (startDateAndTime > endDateAndTime) {
+      setError("Error: end cannot be before start.");
+      return;
+    }
+
+    setEvents([
+      {
+        title: `${formatAMPM(startTime)} ${title}`,
```

Comment:
Why have the starttime in the title?

Reply:
We'll remove this for now and research into what the proper way is to show the start and end time on the Big Calendar UI.